<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>570</width>
    <height>390</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>570</width>
    <height>390</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>570</width>
    <height>390</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>OTA Upload</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QTabWidget" name="twTabs">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>572</width>
      <height>392</height>
     </rect>
    </property>
    <property name="currentIndex">
     <number>0</number>
    </property>
    <widget class="QWidget" name="tab">
     <attribute name="title">
      <string>Upload</string>
     </attribute>
     <widget class="QPushButton" name="bUpload">
      <property name="geometry">
       <rect>
        <x>220</x>
        <y>260</y>
        <width>331</width>
        <height>51</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>11</pointsize>
       </font>
      </property>
      <property name="text">
       <string>UPLOAD</string>
      </property>
     </widget>
     <widget class="QPushButton" name="bChooseImageFile">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>30</y>
        <width>91</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string>choose File</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="leImagePath">
      <property name="geometry">
       <rect>
        <x>120</x>
        <y>30</y>
        <width>431</width>
        <height>31</height>
       </rect>
      </property>
      <property name="readOnly">
       <bool>true</bool>
      </property>
     </widget>
     <widget class="QGroupBox" name="groupBox">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>260</y>
        <width>181</width>
        <height>81</height>
       </rect>
      </property>
      <property name="title">
       <string>Program or SPIFFS</string>
      </property>
      <widget class="QRadioButton" name="rbUploadProgram">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>20</y>
         <width>95</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>Program</string>
       </property>
       <property name="checked">
        <bool>true</bool>
       </property>
      </widget>
      <widget class="QRadioButton" name="rbUploadSpiffs">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>50</y>
         <width>95</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>SPIFFS</string>
       </property>
      </widget>
     </widget>
     <widget class="QLabel" name="label_9">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>9</y>
        <width>111</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Binary Image File</string>
      </property>
     </widget>
     <widget class="QProgressBar" name="pbProgress">
      <property name="geometry">
       <rect>
        <x>220</x>
        <y>320</y>
        <width>331</width>
        <height>31</height>
       </rect>
      </property>
      <property name="value">
       <number>0</number>
      </property>
      <property name="alignment">
       <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
      </property>
      <property name="textVisible">
       <bool>false</bool>
      </property>
     </widget>
     <widget class="QTabWidget" name="twUploadMethod">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>96</y>
        <width>531</width>
        <height>151</height>
       </rect>
      </property>
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="tab_4">
       <attribute name="title">
        <string>Serial</string>
       </attribute>
       <widget class="QSpinBox" name="sbComPort">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>30</y>
          <width>61</width>
          <height>31</height>
         </rect>
        </property>
       </widget>
       <widget class="QLabel" name="label">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>10</y>
          <width>31</width>
          <height>16</height>
         </rect>
        </property>
        <property name="text">
         <string>COM</string>
        </property>
       </widget>
       <widget class="QComboBox" name="cbUploadSpeed">
        <property name="geometry">
         <rect>
          <x>90</x>
          <y>30</y>
          <width>171</width>
          <height>31</height>
         </rect>
        </property>
        <item>
         <property name="text">
          <string>115200</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>256000</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>512000</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>921600</string>
         </property>
        </item>
       </widget>
       <widget class="QLabel" name="label_12">
        <property name="geometry">
         <rect>
          <x>90</x>
          <y>10</y>
          <width>55</width>
          <height>16</height>
         </rect>
        </property>
        <property name="text">
         <string>Speed</string>
        </property>
       </widget>
       <widget class="QCheckBox" name="cbEraseFlash">
        <property name="geometry">
         <rect>
          <x>280</x>
          <y>30</y>
          <width>91</width>
          <height>31</height>
         </rect>
        </property>
        <property name="text">
         <string>Erase Flash</string>
        </property>
       </widget>
      </widget>
      <widget class="QWidget" name="tab_5">
       <attribute name="title">
        <string>IP</string>
       </attribute>
       <widget class="QComboBox" name="cbSelectDevice">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>22</y>
          <width>211</width>
          <height>31</height>
         </rect>
        </property>
       </widget>
       <widget class="QSpinBox" name="sbUploadPort">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>80</y>
          <width>71</width>
          <height>22</height>
         </rect>
        </property>
        <property name="buttonSymbols">
         <enum>QAbstractSpinBox::NoButtons</enum>
        </property>
        <property name="maximum">
         <number>100000</number>
        </property>
        <property name="value">
         <number>8266</number>
        </property>
       </widget>
       <widget class="QLabel" name="label_7">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>60</y>
          <width>55</width>
          <height>16</height>
         </rect>
        </property>
        <property name="text">
         <string>Port</string>
        </property>
       </widget>
       <widget class="QLineEdit" name="leUploadPasswd">
        <property name="geometry">
         <rect>
          <x>100</x>
          <y>80</y>
          <width>161</width>
          <height>22</height>
         </rect>
        </property>
       </widget>
       <widget class="QLabel" name="label_8">
        <property name="geometry">
         <rect>
          <x>100</x>
          <y>60</y>
          <width>61</width>
          <height>16</height>
         </rect>
        </property>
        <property name="text">
         <string>Password</string>
        </property>
       </widget>
       <widget class="QCheckBox" name="cDebug">
        <property name="geometry">
         <rect>
          <x>280</x>
          <y>80</y>
          <width>81</width>
          <height>20</height>
         </rect>
        </property>
        <property name="text">
         <string>DEBUG</string>
        </property>
       </widget>
       <widget class="QLabel" name="label_11">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>3</y>
          <width>55</width>
          <height>16</height>
         </rect>
        </property>
        <property name="text">
         <string>Address</string>
        </property>
       </widget>
      </widget>
     </widget>
     <widget class="QLabel" name="label_10">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>75</y>
        <width>91</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Upload Method</string>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_2">
     <attribute name="title">
      <string>Devices</string>
     </attribute>
     <widget class="QGroupBox" name="groupBox_2">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>230</y>
        <width>530</width>
        <height>121</height>
       </rect>
      </property>
      <property name="title">
       <string>New Device</string>
      </property>
      <widget class="QSpinBox" name="sbNewDeviceIP3">
       <property name="geometry">
        <rect>
         <x>480</x>
         <y>50</y>
         <width>41</width>
         <height>25</height>
        </rect>
       </property>
       <property name="buttonSymbols">
        <enum>QAbstractSpinBox::NoButtons</enum>
       </property>
       <property name="correctionMode">
        <enum>QAbstractSpinBox::CorrectToPreviousValue</enum>
       </property>
       <property name="maximum">
        <number>255</number>
       </property>
       <property name="value">
        <number>0</number>
       </property>
      </widget>
      <widget class="QLabel" name="label_3">
       <property name="geometry">
        <rect>
         <x>423</x>
         <y>60</y>
         <width>16</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>.</string>
       </property>
      </widget>
      <widget class="QPushButton" name="bAddNewDevice">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>80</y>
         <width>161</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>Add Device</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_2">
       <property name="geometry">
        <rect>
         <x>373</x>
         <y>60</y>
         <width>16</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>.</string>
       </property>
      </widget>
      <widget class="QSpinBox" name="sbNewDeviceIP1">
       <property name="geometry">
        <rect>
         <x>380</x>
         <y>50</y>
         <width>41</width>
         <height>25</height>
        </rect>
       </property>
       <property name="buttonSymbols">
        <enum>QAbstractSpinBox::NoButtons</enum>
       </property>
       <property name="maximum">
        <number>255</number>
       </property>
       <property name="value">
        <number>168</number>
       </property>
      </widget>
      <widget class="QSpinBox" name="sbNewDeviceIP0">
       <property name="geometry">
        <rect>
         <x>330</x>
         <y>50</y>
         <width>41</width>
         <height>25</height>
        </rect>
       </property>
       <property name="buttonSymbols">
        <enum>QAbstractSpinBox::NoButtons</enum>
       </property>
       <property name="maximum">
        <number>255</number>
       </property>
       <property name="value">
        <number>192</number>
       </property>
      </widget>
      <widget class="QLineEdit" name="leNewDeviceName">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>50</y>
         <width>311</width>
         <height>25</height>
        </rect>
       </property>
      </widget>
      <widget class="QLabel" name="label_4">
       <property name="geometry">
        <rect>
         <x>473</x>
         <y>60</y>
         <width>16</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>.</string>
       </property>
      </widget>
      <widget class="QSpinBox" name="sbNewDeviceIP2">
       <property name="geometry">
        <rect>
         <x>430</x>
         <y>50</y>
         <width>41</width>
         <height>25</height>
        </rect>
       </property>
       <property name="buttonSymbols">
        <enum>QAbstractSpinBox::NoButtons</enum>
       </property>
       <property name="correctionMode">
        <enum>QAbstractSpinBox::CorrectToPreviousValue</enum>
       </property>
       <property name="maximum">
        <number>255</number>
       </property>
       <property name="value">
        <number>0</number>
       </property>
      </widget>
      <widget class="QLabel" name="label_5">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>30</y>
         <width>55</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>Name</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_6">
       <property name="geometry">
        <rect>
         <x>330</x>
         <y>30</y>
         <width>81</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>IP Adress</string>
       </property>
      </widget>
      <widget class="QPushButton" name="bClearAllDevices">
       <property name="geometry">
        <rect>
         <x>180</x>
         <y>80</y>
         <width>141</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>Clear all Devices</string>
       </property>
      </widget>
     </widget>
     <widget class="QTableWidget" name="twDevices">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>10</y>
        <width>531</width>
        <height>201</height>
       </rect>
      </property>
      <attribute name="horizontalHeaderCascadingSectionResizes">
       <bool>true</bool>
      </attribute>
      <attribute name="horizontalHeaderDefaultSectionSize">
       <number>255</number>
      </attribute>
      <attribute name="horizontalHeaderStretchLastSection">
       <bool>true</bool>
      </attribute>
      <attribute name="verticalHeaderShowSortIndicator" stdset="0">
       <bool>false</bool>
      </attribute>
      <attribute name="verticalHeaderStretchLastSection">
       <bool>false</bool>
      </attribute>
      <column>
       <property name="text">
        <string>Name</string>
       </property>
      </column>
      <column>
       <property name="text">
        <string>IP</string>
       </property>
      </column>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_3">
     <attribute name="title">
      <string>Console</string>
     </attribute>
     <widget class="QPushButton" name="bClearOutput">
      <property name="geometry">
       <rect>
        <x>462</x>
        <y>4</y>
        <width>101</width>
        <height>41</height>
       </rect>
      </property>
      <property name="text">
       <string>Clear output</string>
      </property>
     </widget>
     <widget class="QPlainTextEdit" name="teConsoleOutput">
      <property name="geometry">
       <rect>
        <x>2</x>
        <y>50</y>
        <width>561</width>
        <height>311</height>
       </rect>
      </property>
      <property name="acceptDrops">
       <bool>false</bool>
      </property>
      <property name="textInteractionFlags">
       <set>Qt::NoTextInteraction</set>
      </property>
     </widget>
    </widget>
   </widget>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <tabstops>
  <tabstop>cbSelectDevice</tabstop>
  <tabstop>bChooseImageFile</tabstop>
  <tabstop>rbUploadProgram</tabstop>
  <tabstop>rbUploadSpiffs</tabstop>
  <tabstop>bUpload</tabstop>
  <tabstop>leNewDeviceName</tabstop>
  <tabstop>sbNewDeviceIP0</tabstop>
  <tabstop>sbNewDeviceIP1</tabstop>
  <tabstop>sbNewDeviceIP2</tabstop>
  <tabstop>sbNewDeviceIP3</tabstop>
  <tabstop>bAddNewDevice</tabstop>
  <tabstop>twTabs</tabstop>
  <tabstop>leImagePath</tabstop>
 </tabstops>
 <resources/>
 <connections/>
</ui>
